File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int minDiff = INT_MAX;
        for (int i = 0; i < arr.size() - 1; i++) {
            if (abs(arr[i + 1] - arr[i]) < minDiff) {
                minDiff = abs(arr[i + 1] - arr[i]);
            }
        }

        vector<vector<int>> ans;
        for (int i = 0; i < arr.size() - 1; i++) {
            if (abs(arr[i + 1] - arr[i]) == minDiff) {
                int a = min(arr[i + 1], arr[i]);
                int b = max(arr[i + 1], arr[i]);
                ans.push_back({a, b});
            }
        }
        return ans;
    }
};

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.go
func minimumAbsDifference(arr []int) [][]int {
    sort.Ints(arr)
    ans := [][]int{}
    minDiff := arr[1] - arr[0]
    for i:= 1 ; i< len(arr) ; i++ {
        if minDiff > arr[i] - arr[i-1] {
            minDiff = arr[i] - arr[i-1]
        }
    }
    for i:= 1 ; i< len(arr) ; i++ {
           if minDiff == arr[i] -arr[i-1] {
             	ans = append(ans, []int{arr[i-1], arr[i]})
           }
        
    }
    return ans
}